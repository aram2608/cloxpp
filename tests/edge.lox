print "== precedence ==";
print 1 + 2 * 3 == 7;     // true  ( * binds tighter than +, == after )
print !(false == false);  // false (== before ! )

print "== dangling else (associates to nearest if) ==";
var x = 0;
if (true)
  if (false)
    x = 1;
  else
    x = 2;
print x;                  // 2

print "== chained if/else with blocks ==";
if (false) {
  print "no";
} else if (false) {
  print "no";
} else {
  print "yes";            // yes
}

print "== logical short-circuit (no side effects) ==";
var t = 0;
fun bump() { t = t + 1; print "bump"; }
false and bump();         // should NOT print "bump"
true or bump();           // should NOT print "bump"
print t;                  // 0
