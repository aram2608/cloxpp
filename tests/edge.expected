== precedence ==
true
false
== dangling else (associates to nearest if) ==
2.000000
== chained if/else with blocks ==
yes
== logical short-circuit (no side effects) ==
0.000000
